### ISSUE TITLE
[cmp.alg] Notes for ill-formed cases are inconsistent with those in [range.access]

### ISSUE THREAD

**Issue by frederick-vs-ja** [cplusplus/draft#4492]
It's noted that "This case can result in substitution failure when `*_­order(E, F)` appears in the immediate context of a template instantiation." in [[cmp.alg]](https://eel.is/c++draft/cmp.alg), which implies that only one of the two ill-formed cases results in substitution failure. And such notes don't exist in specifications of `compare_*_order_fallback` CPOs.

Are the cases for hard error overriding [[customization.point.object]/4](https://eel.is/c++draft/customization.point.object#4) are intended?

If not, I think that we should also use the form "(Diagnosable) ill-formed cases above result in substitution failure when `*(E, F)` appears in the immediate context of a template instantiation" (from [[range.access]](https://eel.is/c++draft/range.access)) in [cmp.alg] and apply the notes for all six CPOs.

By the way, the uses of `std::` are also inconsistent in these notes currently.

**Comment by jensmaurer** [comment]
So, the suggestion is to replace the notes in [cmp.alg] about immediate context with the phrasing from [ranges.access]: "Ill-formed cases above result in substitution failure when strong_order(E) appears in the immediate context of a template instantiation." so that the note covers both ill-formed cases, not just the last one.

Also, it feels that "performed in a context that does not include a declaration of" should be replaced by the more precise "performed in a context in which unqualified lookup for strong_order finds nothing".

@jwakely, are these changes (in particular the first one) net improvements, and sufficiently editorial?

**Comment by jwakely** [comment]
> which implies that only one of the two ill-formed cases results in substitution failure.

It might imply it, but it's not normative, so I don't think the presence or absence of such notes can override anything normative. But I'm not sure whether [customization.point.object] p4 applies to the first bullet; is "If the decayed types of E and F differ, strong_­order(E, F) is ill-formed." a "requirement specified in the customization point's definition"? It's not stated as a requirement.

When I implemented `std::strong_order` I used a `static_assert` to check the decayed types are the same, which means it doesn't result in substitution failure. I'd like to hear from @BRevzin whether the intention is that all the ill-formed cases in [cmp.alg] are supposed to result in substitution failures rather than hard errors.
